components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-list.jsp

```diff
@@ -30,166 +49,279 @@
-                        location.assign("idp-mgt-list-load.jsp");
+   }
+
+   function deleteIdPName(idpName,pageNumberInt,filterString) {
+      function doDelete() {
+      $.ajax({
+      type: 'POST',
+      url: 'idp-mgt-delete-finish-ajaxprocessor.jsp',
+      headers: {
+        Accept: "text/html"
+      },
+      data: 'idPName=' + encodeURIComponent(idpName),
+      async: false,
+      success: function (responseText, status) {
+         if (status == "success") {
+            location.assign("idp-mgt-list.jsp?pageNumber=" + pageNumberInt.toString() +
+            "&region=region1&item=idp_list&filterString=" + filterString);
+         }
+      }
+      });
+      }
+       CARBON.showConfirmationDialog('Are you sure you want to delete "'  + idpName + '" IdP information?', doDelete, null);
+   }
+
+   function enableOrDisableIdP(idpName, indicator) {
+      $.ajax({
+        type: 'POST',
+        url: 'idp-mgt-edit-finish-ajaxprocessor.jsp',
+        headers: {
+           Accept: "text/html"
+        },
+        data: 'idPName=' + encodeURIComponent(idpName) + '&enable=' + indicator,
+        async: false,
+        success: function (responseText, status) {
+
+        if (status == "success") {
+           location.assign("idp-mgt-list.jsp");
+        }
+        }
+        });
+   }
+</script>
+<fmt:bundle
+   basename="org.wso2.carbon.idp.mgt.ui.i18n.Resources">
+   <carbon:breadcrumb label="application.mgt"
+      resourceBundle="org.wso2.carbon.idp.mgt.ui.i18n.Resources"
+      topPage="true" request="<%=request%>"/>
+   <div id="middle">
+      <h2>
+         <fmt:message key='identity.providers'/>
+      </h2>
+      <div id="workArea">
+         <%
+            final String IDP_NAME_FILTER = "filterString";
+            final String DEFAULT_FILTER = "*";
+            String BUNDLE = "org.wso2.carbon.idp.mgt.ui.i18n.Resources";
+            ResourceBundle resourceBundle = ResourceBundle.getBundle(BUNDLE, request.getLocale());
+
+            IdentityProvider[] idpsToDisplay = new IdentityProvider[0];
+            boolean hasFilter = false;
+            String paginationValue = "region=region1&item=idp_list";
+            String filterString = request.getParameter(IDP_NAME_FILTER);
+
+            if (StringUtils.isBlank(filterString)) {
+                filterString = "*";
+            } else {
+                filterString = filterString.trim();
+                paginationValue = "region=region1&item=idp_list&filterString=" + filterString;
+                hasFilter = true;
+            }
+
+            int pageNumberInt = 0;
+            int numberOfPages = 0;
+            int resultsPerPageInt = 15;
+            String resultsPerPage = IdentityUtil.getProperty("DefaultItemsPerPage");
+            String pageNumber = request.getParameter("pageNumber");
+
+            if (StringUtils.isNotBlank(resultsPerPage)) {
+                try {
+                    resultsPerPageInt = Integer.parseInt(resultsPerPage);
```

Comment:
we need to assign the default value if the user has configured a negative value in the config file